### PR TITLE
Pipeline: Enable OpenRewrite checks, separate modernizer check, and checkstyle comments

### PR DESCRIPTION
**Pull request by koppor**
This PR changes four things:

### Rename "Checkstyle" to "Style checks"

image

### OpenRewrite Checks

Fixes https://github.com/JabRef/jabref-issue-melting-pot/issues/163

image

image

### Checkstyle comment on PR

Fixes https://github.com/JabRef/jabref-issue-melting-pot/issues/164

image

We need two empty lines in YAML. See https://stackoverflow.com/a/21699210/873282.

### Separate modernizer check

Included in "Style checks" - and fails if modernizer has something to say.

---

```[tasklist]
### Compulsory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
```

### CODE REVIEW

**Comment by koppor** [comment]
Blocked by https://github.com/JabRef/jabref/pull/9859

**Review by calixtus — APPROVED** [review]

**Comment by Siedlerchr** [comment]
Suggestion: Rename to "Code style check"

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -34,15 +34,43 @@ jobs:
...
+        if: ${{ failure() }}
+        uses: thollander/actions-comment-pull-request@v2
+        with:
+          message: >
+              GitHub has automatic checks in place.
+              One of them is checking for the formatting of the code.
+              Consistent formatting ensures more easy reading of the code.
+              Thus, we pay attention that JabRef's code follows the same code style.
+
+
```
Comment (Siedlerchr) [inline .github/workflows/tests.yml:56]:
```suggestion
          


```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -34,15 +34,43 @@ jobs:
...
+              Your code currently fails to meet JabRef's code guidelines.
```
Comment (Siedlerchr) [inline .github/workflows/tests.yml:57]:
```suggestion
              Your code currently does not meet JabRef's code guidelines.
```

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -34,15 +34,43 @@ jobs:
...
+              The tool [reviewdog](https://github.com/reviewdog/reviewdog#readme) already placed comments on GitHub to indicate the places.
```
Comment (Siedlerchr) [inline .github/workflows/tests.yml:58]:
```suggestion
              The tool [reviewdog](https://github.com/reviewdog/reviewdog#readme) already placed comments on GitHub to indicate the places. See the tab "Files" in you PR.
```

**Pull request merged**